src/protocol/libp2p/identify.rs

```diff
@@ -98,20 +148,27 @@ impl Config {
         protocol_version: String,
         user_agent: Option<String>,
         public_addresses: Vec<Multiaddr>,
-    ) -> (Self, Box<dyn Stream<Item = IdentifyEvent> + Send + Unpin>) {
+    ) -> (
+        Self,
+        IdentifyPublicAddresses,
+        Box<dyn Stream<Item = IdentifyEvent> + Send + Unpin>,
```

Comment:
This could return `IdentifyHandle` which would hold `IdentifyPublicAddresses` and would implement `Stream<Item = IdentifyEvent>`, following the design of other protocols.

Reply:
I like that idea, will certainly implement it! Thanks for the review!

Have put this on pending shortly, I have a hunch we'll need to expose an `ExternalAddrConfirmed` (like libp2p) to suffice other usecases (https://github.com/paritytech/litep2p/issues/211). Will wait for Dmitry to confirm that these are needed for the bootnodes on DHT🙏

Reply:
Sorry to have not responded here. We discussed https://github.com/paritytech/litep2p/issues/211 with @lexnv, and my idea was to introduce a globally accessible list of public addresses in litep2p. If some protocol needs external addresses, it can fetch them. This includes `Identify` and `Kademlia` protocols now.

As a first step, we can manually notify `litep2p` about confirmed external addresses from the client code. As a second step, we can move the heuristic of confirming external addresses seen by three or more peers from `polkadot-sdk` to `litep2p`. I have updated https://github.com/paritytech/litep2p/issues/211 to be more clear about this.

I don't think `ExternalAddrConfirmed` event will help with providing up-to-date peer addresses for local content providers in Kademlia from within `litep2p`, but the option to get addresses from the global list will.

Reply:
Your plan sounds good. The easiest place to store the external addresses is probably `TransportManager`, partly since there already exists code for storing known addresses via `TransportManagerHandle` and since every protocol has a handle to `TransportManager`, it should require little architectural changes.